text/metrics/0146-metrics-prototype-scenarios.md

```diff
@@ -0,0 +1,220 @@
+  specification, with multiple language SIGs providing RC (release candidate) or
+  [stable](https://github.com/open-telemetry/opentelemetry-specification/blob/9047c91412d3d4b7f28b0f7346d8c5034b509849/specification/versioning-and-stability.md#stable)
+  clients.
+
+In this document, we will focus on two scenarios that we use for prototyping
+metrics API/SDK. The goal is to have two scenarios which clearly capture the
+major requirements, so we can work with language client SIGs to prototype,
+gather the learnings, determine the scopes and stages. Later the scenarios can
+be used as examples and test cases for all the language clients.
+
+Here are the languages we've agreed to use during the prototyping:
+
+* C#
+* Java
+* Python
+
+Instead of boiling the ocean, we will need to divide the work into multiple
+stages:
+
+1. Do the end-to-end prototype to get the overall understanding of the problem
+   domain. We should also clarify the scope and be able to articulate it
+   precisely during this stage, here are some examples:
+
+    * Why do we want to introduce brand new metrics APIs versus taking a well
+      established API (e.g. Prometheus and Micrometer), what makes OpenTelemetry
+      metrics API different (e.g. Baggage)?
+    * Do we need to consider OpenCensus Stats API shim, or this is out of scope?
+
+2. Focus on a core subset of API, cover the end-to-end library instrumentation
+   scenario. At this stage we don't expect to cover all the APIs as some of them
+   might be very similar (e.g. if we know how to record an integer, we don't
+   have to work on float/double as we can add them later by replicating what
+   we've done for integer).
+
+3. Focus on a core subset of SDK. This would help us to get the end-to-end
+   application.
+
+4. Replicate stage 2 to cover the complete set of APIs.
+
+5. Replicate stage 4 to cover the complete set of SDKs.
+
+## Scenario 1: Grocery
+
+The **Grocery** scenario covers how a developer could use metrics API and SDK in
+a final application. It is a self-contained application which covers:
+
+* How to instrument the code in a vendor agnostic way
+* How to configure the SDK and exporter
+
+Considering there might be multiple grocery stores, the metrics we collect will
+have the store name as a dimension - which is fairly static (not changing while
+the store is running).
+
+The store has plenty supply of potato and tomato, with the following price:
+
+* Potato: $1.0 / ea
+* Tomato: $3.0 / ea
+
+Each customer has a unique name (e.g. customerA, customerB), a customer could
+come to the store multiple times. Here goes the Python snippet:
+
+```python
+store = GroceryStore("Portland")
+store.process_order("customerA", {"potato": 2, "tomato": 3})
+store.process_order("customerB", {"tomato": 10})
+store.process_order("customerC", {"potato": 2})
+store.process_order("customerA", {"tomato": 1})
+```
+
+When the store is closed, we will report the following metrics:
```

Comment:
I will assume you mean we instrument enough data so that a backend system can generate the report per day when the store is closed?  I don't think you mean we only send metric ONCE at end of day.

Reply:
Yep, this is similar to what @cijothomas mentioned [here](https://github.com/open-telemetry/oteps/pull/146/files#r578138526), let me change it to a per minute report.